### PR TITLE
Agent orchestrator settings and Azure AI Direct orchestrator type implementation

### PR DESCRIPTION
**Pull request by joelhulen**
# Agent orchestrator settings and Azure AI Direct orchestrator type implementation

## Details on the issue fix or feature implementation

This update provides a flexible approach for supplying orchestrator settings for an agent. It also implements a new orchestrator type (`AzureAIDirect`) that enables an agent to call LLMs deployed by Azure AI Studio.

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have included inline docs for my changes, where applicable
- [x]  I have successfully run a local build
- [ ]  I have provided the required update scripts, where applicable
- [ ]  I have updated relevant docs, where applicable

> [!NOTE]
> Instead of adding `X`'s inside the checkboxes you wish to check above, first submit the PR, then check the boxes in the rendered description.

### CODE REVIEW

File: src/dotnet/Common/Models/Orchestration/OrchestrationRequest.cs
Comment (ciprianjichici) [inline src/dotnet/Common/Models/Orchestration/OrchestrationRequest.cs:1]: The agent name should be a property in `OrchestrationSettings` not a direct property of the orchestration request.

File: src/dotnet/Common/Models/Orchestration/OrchestrationSettings.cs
Comment (ciprianjichici) [inline src/dotnet/Common/Models/Orchestration/OrchestrationSettings.cs:1]: The agent name should be part of `OrchestrationSettings`, as the agent is an intrinsic part of the orchestration process.

File: src/dotnet/Common/Services/ResourceProviders/ResourceProviderServiceBase.cs
Comment (ciprianjichici) [inline src/dotnet/Common/Services/ResourceProviders/ResourceProviderServiceBase.cs:1]: This duplicates logic that is already in `GetResourceInstancesFromPath`. Using `GetResourcesInstancesFromPath(objectId)[4..]` provides the same result.

File: src/dotnet/AgentFactory/Services/AzureAIDirectService.cs
Comment (ciprianjichici) [inline src/dotnet/AgentFactory/Services/AzureAIDirectService.cs:1]: Error messages are still referencing LangChain.

Comment (kylebunting) [reply src/dotnet/AgentFactory/Services/AzureAIDirectService.cs:1]: Fixed

**Review by ciprianjichici — APPROVED** [review]

**Pull request merged**